src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala

```
@@ -127,6 +128,29 @@
     thurloeDao.saveKeyValues(userInfo, profilePropertyMap)
   }
 
+  private def unlinkNihAccount(userInfo: UserInfo): Future[Unit] = {
+    val nihKeys = Set("linkedNihUsername", "linkExpireTime")
+
+    Future.traverse(nihKeys) { nihKey =>
+      thurloeDao.deleteKeyValue(userInfo.id, nihKey, userInfo)
+    } map { results =>
+      if(!results.forall(_.isSuccess))
+        throw new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, "Unable to unlink NIH account"))
```

It looks like this is swallowing the exception thrown by `ThurloeDao.deleteKeyValue`, and will lose the key that failed to be deleted.

---

Addressed in 7d4194a6dc0d60f90e49aebff2cc6afb5d9e0702

---

Thanks so much!